scanner/src/test/kotlin/scanners/fossid/FossIdTest.kt

```diff
@@ -0,0 +1,829 @@
+            val vcsInfo = createVcsInfo()
+
+            val service = config.createService()
+                .expectProjectRequest(projectCode)
+                .expectListScans(projectCode, emptyList())
+                .expectCheckScanStatus(scanCode, ScanState.NOT_STARTED, ScanState.SCANNING, ScanState.FINISHED)
+                .expectCreateScan(projectCode, scanCode, vcsInfo)
+                .expectDownload(scanCode)
+                .expectDeleteScan(scanCode)
+                .mockFiles(scanCode)
+            coEvery { service.runScan(USER, API_KEY, scanCode) } returns EntityResponseBody(status = 200)
+
+            val fossId = createFossId(config)
+
+            shouldThrow<TimeoutCancellationException> {
+                withTimeout(1000) {
+                    fossId.scanPackages(listOf(createPackage(createIdentifier(index = 1), vcsInfo)))
+                }
+            }
+        }
+
+        "support triggering asynchronous scans" {
+            val pkgId = createIdentifier(index = 1)
+            val projectCode = projectCode(PROJECT)
+            val scanCode = scanCode(PROJECT, null)
+            val config = createConfig(deltaScans = false, waitForResult = false)
+            val vcsInfo = createVcsInfo()
+
+            val service = config.createService()
+                .expectProjectRequest(projectCode)
+                .expectListScans(projectCode, emptyList())
+                .expectCheckScanStatus(scanCode, ScanState.NOT_STARTED, ScanState.SCANNING, ScanState.FINISHED)
+                .expectCreateScan(projectCode, scanCode, vcsInfo)
+                .expectDownload(scanCode)
+
+            val fossId = createFossId(config)
+
+            val scannerRun = fossId.scan(listOf(createPackage(pkgId, vcsInfo)))
+
+            scannerRun.results.collectIssues()[pkgId] shouldNotBeNull {
+                this shouldHaveSize 1
+                val issue = first()
+                issue.source shouldBe pkgId.toCoordinates()
+                issue.severity shouldBe Severity.HINT
+                issue.message shouldContain "asynchronous mode"
+            }
+
+            coVerify(exactly = 0) {
+                service.checkScanStatus(USER, API_KEY, any())
+            }
+        }
+
+        "apply filters for namespaces and authors" {
+            val id1 = createIdentifier(1)
+            val pkg1 = createPackage(id1, createVcsInfo())
+
+            val ignoredAuthor = "hacker"
+            val pkgAuthors = setOf("foo", "bar", ignoredAuthor)
+            val id2 = createIdentifier(2).copy(namespace = "anotherNamespace")
+            val pkg2 = createPackage(id2, createVcsInfo("anotherProject"), authors = pkgAuthors)
+
+            val config = createConfig(packageNamespaceFilter = id1.namespace, packageAuthorsFilter = ignoredAuthor)
+            val fossId = createFossId(config)
+
+            val scannerRun = fossId.scan(listOf(pkg1, pkg2))
+            scannerRun.results.collectIssues().keys should beEmpty()
```

Comment:
Unfortunately, checking that is not enough. How do you know that `pkg1` was not filtered and scanned ?

Reply:
oh ok. Maybe a comment then ? I let you judge !